portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java

```diff
@@ -423,6 +424,62 @@ public void clearUserRoles(long userId) throws PortalException {
 		reindex(userId);
 	}
 
+	@Override
+	public Role copyRole(
+			Role sourceRole, String duplicateRoleName,
+			ServiceContext serviceContext)
+		throws PortalException {
+
+		Role duplicateRole = addRole(
+			null, sourceRole.getUserId(), sourceRole.getClassName(), 0,
```

Comment:
the userId should be the one passed as param because otherwise seems to be created by the original user that is not true